### PR TITLE
feat(actions): add input in publish image action and default it to ghcr.io

### LINKED ISSUE #1140: Publish image CI action has undeclared dependency.
`.github/actions/publish-image` uses environment variable `$URL_DOCKER_REGISTRY` to decide which root registry to push to.   
If this is not set globally, or at some step before the action's invocation, it will fail to publish. 

This can be surprising, as the necessity of this environment variable is not mentioned anywhere. 

It would be better to have this as an input to the action, with a default value: 'ghcr.io'.

### PR DESCRIPTION
**Pull request by PanGan21**
- Adds `url-docker-registry` input in publish image github action with default value to `ghcr.io`
- Pass parameter where publish image action is called

Closes #1140

### CODE REVIEW

File: .github/actions/build-image/action.yml
Comment (nain-F49FF806) [inline .github/actions/build-image/action.yml:95]: This seems like the right way to go. 
But note that now we will again have a surprising fail scenario if this environment variable is not set. 

Let's repeat the earlier process and add *url-docker-registry* as explicit input of this action, and then pass it down the chain.

**Review by nain-F49FF806 — CHANGES_REQUESTED** [review]
Thank you for making the PR! This looks good. You have alleviated the main concern of #1140.
I have only a few minor additions/edits I would like to see before we merge it in. It should take you only a couple of minutes more. Please take a look through the review.

File: .github/actions/publish-image/action.yml
Comment (PanGan21) [reply .github/actions/publish-image/action.yml:1]: Updated!

File: .github/actions/build-image/action.yml
Comment (PanGan21) [reply .github/actions/build-image/action.yml]: Done! Thanks a lot for the feedback! 
I still kept the same pattern, optional with default value. Let me know if that works!

**Review by nain-F49FF806 — APPROVED** [review]

**Comment by nain-F49FF806** [comment]
LGTM. 

I'll let this be open until Monday to let other maintainers have a look. 
Thank you for the contribution!

**Review by Patrik-Stas — APPROVED** [review]

**Pull request merged**
